src/dsp_tools/utils/connection_live.py

```diff
@@ -240,13 +239,12 @@ def post(
         if not route.startswith("/"):
             route = f"/{route}"
         url = self.server + route
-        headers = {}
         if jsondata:
-            headers["Content-Type"] = "application/json; charset=UTF-8"
+            self.headers["Content-Type"] = "application/json; charset=UTF-8"
```

Comment:
I wouldn't store `headers` as a class attribute. Because once you make a POST request with jsondata, the header for ALL SUBSEQUENT requests will be "Content-Type" = "application/json; charset=UTF-8". So also for subsequent GET requests, and POST requests without jsondata. 

A cleaner solution would be to pass the "X-Asset-Ingested" really only for this one POST request that creates the resource.

Reply:
In this case, either I would have to have a attribute solely about the ingest, which also does not seem like a good solution or write another post function...?

Reply:
@BalduinLandolt you worked on this a lot. Do you have a suggestion?

Reply:
I would suggest adding the param `headers: dict[str, str] | None = None` to *all* methods (Get, Put, Post, Delete) both on the protocol and the implementation, and add it to the default headers, if it is defined; with the default value, it shouldn't break any existing code, but it would allow for adding any header anywhere if needed, which gives us a lot of flexibility also for the future.  
But in principal it wouldn't break anything to do it the way it is now - superfluous headers are just ignored. However, I don't think that's the most sustainable solution in the long run: We may run into a situation where we want to be more specific on when we send which header.

Reply:
Thanks for the clarification, Balduin, that's what I meant. I strongly vote for Balduin's solution.

Reply:
Okay, that's fine. I just approved. You can then do the header in a separate PR.

Reply:
Already made a ticket